feat: person search endpoint

jennifer-richards opened the pull request:
This is the datatracker side of https://github.com/ietf-tools/purple/pull/365. Adds a paginated search endpoint for Persons.

I'm conflicted on naming. The names here (`RpcPerson*`) follow the pattern we've been using with `rpc` prefixes, but have the decidedly awful result that the purple view works with something called an `RpcPerson` that is not purple's `rpc.models.RpcPerson`. Suggestions welcome.

Note that this is starting to use DRF for the datatracker's purple interface. We have so far only used that for things supporting red. The motivation here is not having to implement custom pagination. The schema changes to `rpcapi.yaml` were done by generating a schema via spectacular and manually pulling in the relevant bits. We will want to make that process better, but I think that's a separate project.

jennifer-richards committed:
feat: person search endpoint

kesara commented on ietf/api/views_rpc.py:
```diff
@@ -63,6 +66,35 @@ def rpc_persons(request):
+    max_limit = 100
+
+
+class RpcPersonSerializer(serializers.ModelSerializer):
```
NIT: This should be in the `api/serializers.py` or `api/serializers_rpc.py`.

kesara commented on rpcapi.yaml:
```diff
@@ -457,6 +487,43 @@ components:
+                    items:
+                        $ref: '#/components/schemas/RpcPerson'
+
+        RpcPerson:
```
You can avoid defining a new schema by using `Person` model.

jennifer-richards replied:
Did it this way to reduce mixing generated and hand-written schema parts, but the schemas are compatible so you're right, we should be optimistic. :-) This also avoids leaking the `RpcPerson` serializer name into the API.

jennifer-richards commented on ietf/api/urls_rpc.py:
```diff
@@ -16,6 +16,7 @@
     url(r"^person/persons_by_email/$", views_rpc.persons_by_email),
     url(r"^person/(?P<person_id>[0-9]+)/$", views_rpc.rpc_person),
     url(r"^persons/$", views_rpc.rpc_persons),
+    url(r"^persons/search/", views_rpc.RpcPersonSearch.as_view()),
```
I think so, but I didn't want to violate the convention we've been using so far in this PR. My inclination is to do a follow-up PR that refactors a bunch of this, including getting rid of the hand-crufted `rpcapi.yaml`.

These names aren't exposed through the API schema so the impacts both of this being a bad name and of changing the names in a later PR shouldn't impact the purple API client. We should be able to match the existing schema with a generated one, at least pretty closely.

jennifer-richards replied:
(Having the prefix in the method/class names is nice if you want to do a `from views_rpc import rpc_whatever` but I don't think we're likely to do it that way.)

kesara replied:
(not relevant to this PR), should `persons_by_email` should be `rpc_persons_by_email`.
What I struggle with is difference between rpc prefixed person and non-rpc prefixed person in `views_rpc`.

jennifer-richards replied:
I guess we weren't as consistent as I thought at a glance. On the datatracker side, there's no such thing as an RpcPerson except where the Rpc is a prefix.

Purple has an `RpcPerson` model that represents someone who has a role with the RPC - editor, manager, etc.

jennifer-richards committed:
refactor: address review comments

  * improved naming of operation/components in API schema
  * reused Person schema component
  * added serializers_rpc.py

jennifer-richards committed:
chore: RpcPersonSerializer -> PersonSerializer

Better matches the hand-written schema.

jennifer-richards committed:
fix: search for entire term, not word-by-word

jennifer-richards committed:
fix: only look at name/plain in search

Including ascii / ascii_short might be useful eventually, but since we only show plain_name in the response it can cause confusing results.

By the same reasoning we could remove email__address as well, but that's useful and I expect we'll include email addresses in our response soon anyway.
